### PR TITLE
Issue #598 layered hfb error

### LINKED ISSUE #598: Support multiple layers in the HFB layered HFB-classes
In #556 Layered HFB packages have been added. A user can define multiple linestring and per linestring a layer can be specified, However the layer specified for a linestring is applied to all linestrings

```
barrier_gdf = gpd.GeoDataFrame(
  geometry=
    [
      linestring1,
      linestring2
    ],
  data=
    {
      "hydraulic_characteristic": [1e-3, 1e-3,],
      "layer": [1, 2,]
    },
)


hfb = imod.mf6.LayeredHorizontalFlowBarrierHydraulicCharacteristic(barrier_gdf)
```

In the snippet above I would expect that linestring1 is applied to layer 1 and that linestring2 is applied to layer 2. However linestring1 is applied to layers 1 and 2 and the same applies for linestrin2

### PR DESCRIPTION
**Pull request by JoerivanEngelen**
Fixes #598

# Description
There was some confusion (also by me) about the use of the ``LayeredHorizontalFlowBarrier`` classes, so @HendrikKok explained me they were intended for single layers. I therefore changed the following to clarify:

- Rename to ``SingleLayerHorizontalFlowBarrier`` classes
- Throw ``ValidationError`` if multiple layers are added to dataset
- Added missing test for SingleLayerHorizontalFlowBarrier class

This means multiple SingleLayerHorizontalFlowBarrier need to be added to GroundwaterFlowModel object, whereas MODFLOW 6 only accepts one single HFB. So these have to be merged together. I think the best opportunity for this is to merge the low level ``Mf6HorizontalFlowBarrier`` objects.

# Checklist
- [x] Links to correct issue
- [x] Update changelog, if changes affect users
- [x] PR title starts with ``Issue #nr``, e.g. ``Issue #737``
- [x] Unit tests were added
- [ ] **If feature added**: Added/extended example

### CODE REVIEW

File: imod/schemata.py [hunk 1]
```diff
@@ -503,6 +503,21 @@ def validate(self, obj: GridDataArray, **kwargs) -> None:
             )
 
 
+class NUniqueValueSchema(BaseSchema):
+    """
+    Schema to validate if amount of unique values is correct.
```
Comment (luitjansl) [inline imod/schemata.py:508]: It appears that the validate method reports an error when the number of unique values exceeds the indicated amount ("length"), but not when it has less unique values than the indicated amount. Please modify the docstring to stress that.

File: imod/schemata.py [hunk 2]
```diff
@@ -503,6 +503,21 @@ def validate(self, obj: GridDataArray, **kwargs) -> None:
...
+    """
+
+    def __init__(self, length: int):
```
Comment (luitjansl) [inline imod/schemata.py:511]: length ->max_unique_values

File: imod/mf6/hfb.py [hunk 3]
```diff
@@ -1105,4 +1120,4 @@ def from_imod5_dataset(cls, imod5_data: dict[str, dict[str, GridDataArray]]):
             dataframe_ls.append(geometry_layer)
         compound_dataframe = pd.concat(dataframe_ls, ignore_index=True)
 
-        return LayeredHorizontalFlowBarrierResistance(compound_dataframe)
+        return cls(compound_dataframe)
```
Comment (luitjansl) [inline imod/mf6/hfb.py:1123]: there is also a "LayeredHorizontalFlowBarrierHydraulicCharacteristic". 
Should the changes in this PR also be applied there?

**Review by luitjansl — COMMENTED** [review]
looks good, some comments

File: imod/mf6/hfb.py (hunk 3 above)
Comment (JoerivanEngelen) [reply imod/mf6/hfb.py:1123]: I think not. The hydraulic characteristic is a MODFLOW 6 concept: it is the inverse of the resistance. In the end, both the ``Resistance`` as well as the ``HydraulicCharacteristic`` variant are converted to a hydraulic characteristic for the Modflow6 HFB package. iMOD5 only allows you to specify a hfb with a resistance, so I don't see a real use case for adding a ``from_imod5_data`` method to the HydraulicCharacteristic class.

File: imod/schemata.py (hunk 2 above)
Comment (JoerivanEngelen) [reply imod/schemata.py:511]: Good one

File: imod/schemata.py (hunk 1 above)
Comment (JoerivanEngelen) [reply imod/schemata.py:508]: Good one, will improve!

**Review by luitjansl — APPROVED** [review]

**Pull request merged**